source/agora/consensus/EnrollmentManager.d

```diff
@@ -190,6 +190,36 @@ public class EnrollmentManager
         return this.validator_set.getEnrolledHeight(enroll_hash);
     }
 
+    /***************************************************************************
+
+        Add a enrollment data to the enrollment pool
+
+        Params:
+            enroll = the enrollment data to add
+            block_height = current block height
+            finder = the delegate to find UTXOs with
+
+        Returns:
+            true if the enrollment data has been added to the enrollment pool
+
+    ***************************************************************************/
+
+    public bool addEnrollment (const ref Enrollment enroll, Height block_height,
```

Comment:
If this function should always be used instead of `pool.add`, then we need to remove the `pool()` getter in this class.

Reply:
> If this function should always be used instead of `pool.add`, then we need to remove the `pool()` getter in this class.

The `ValdiatorSet` and the `EnrollmentPool` are needed at the same time for the operation, so it might be better for the `EnrollmentManager` to have the function. I'm not sure which is better. Do you think putting the code in the Ledger would be better? Anyway, if removing the pool() getter is needed, I'll do it.

Reply:
I'm saying what if the code calls `pool.add` instead of `EnrollmentManager.addEnrollment` by mistake? There is additional logic now in the enrollment manager which is not in the pool.

Reply:
> I'm saying what if the code calls `pool.add` instead of `EnrollmentManager.addEnrollment` by mistake? There is additional logic now in the enrollment manager which is not in the pool.

Right, the `pool()` getter will be removed.